gameplan/extends/client.py

```diff
@@ -5,21 +5,36 @@
 import frappe
 from frappe.model.base_document import get_controller
 
-
 @frappe.whitelist()
-def get_list(doctype=None, fields=None, filters=None, order_by=None, start=0, limit=20, group_by=None, parent=None, debug=False):
-	check_permissions(doctype, parent)
-	query = frappe.qb.get_query(
-		table=doctype,
-		fields=fields,
-		filters=filters,
-		order_by=order_by,
-		offset=start,
-		limit=limit,
-		group_by=group_by,
+def get_list(doctype, **kwargs):
+	controller = get_controller(doctype)
+	if hasattr(controller, "get_list_query"):
+		check_permissions(doctype, kwargs.get("parent"))
+		query = frappe.qb.get_query(
+			table=doctype,
+			fields=kwargs.get("fields"),
+			filters=kwargs.get("filters"),
+			order_by=kwargs.get("order_by"),
+			offset=kwargs.get("start"),
+			limit=kwargs.get("limit", 20),
```

Comment:
Note how the signature differs from `frappe` framework. But remediating this seems to be a more involved refactoring.